unit-tests/src/test/java/org/eclipse/collections/impl/block/procedure/AtomicCountProcedureTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.block.procedure;
+
+import org.eclipse.collections.api.block.predicate.Predicate;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class AtomicCountProcedureTest
+{
+    private static final int STRING_LENGTH = 5;
+    private static final int ZERO = STRING_LENGTH - 5;
+
+    @Test
+    public void getCount()
+    {
+        AtomicCountProcedure<String> atomicCountProcedure = new AtomicCountProcedure<>((Predicate<String>) each ->
```

Comment:
Hi @hongliu2020, I would expect the cast to predicate here to be unnecessary. Can you try removing it?